diff -rN -u configure configure
--- configure	2007-09-16 20:46:15.000000000 +0200
+++ configure	2007-09-16 20:46:17.000000000 +0200
@@ -1470,6 +1470,8 @@
   --with-macapi ((asio/core/sm) default=core)
   --with-asiodir (default=/usr/local/asiosdk2)
   --with-dxdir (default=/usr/local/dx7sdk)
+  --with-dx-includes
+  --with-dx-libraries
   --with-gnu-ld           assume the C compiler uses GNU ld [default=no]
   --with-pic              try to use only PIC/non-PIC objects [default=use
                           both]
@@ -21288,24 +21309,49 @@
 
   mingw* )
 
+        OTHER_OBJS="src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o";
+        LIBS="-lwinmm -lm"
+        PADLL="portaudio.dll";
+        THREAD_CFLAGS="-mthreads"
+        SHARED_FLAGS="-shared";
+        DLL_LIBS="-lwinmm $DLL_LIBS";
+        CFLAGS="$CFLAGS -I\$(top_srcdir)/src/os/win";
+
         echo "WINAPI: $with_winapi"
-        if [ $with_winapi = "directx" ] ; then
-            if [ $with_dxdir ] ; then
-              DXDIR="$with_dxdir";
-            else
-              DXDIR="/usr/local/dx7sdk";
-            fi
-            echo "DXDIR: $DXDIR"
-            OTHER_OBJS="src/hostapi/dsound/pa_win_ds.o src/hostapi/dsound/pa_win_ds_dynlink.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o";
-            LIBS="-lwinmm -lm -ldsound -lole32";
-            PADLL="portaudio.dll";
-	    THREAD_CFLAGS="-mthreads"
-            SHARED_FLAGS="-shared";
-            DLL_LIBS="${DLL_LIBS} -lwinmm -lm -L./dx7sdk/lib -ldsound -lole32";
+
+        if [ $with_dx_includes ] ; then
+          DXINCDIR="$with_dx_includes"
+        elif [ $with_dxdir ] ; then
+          DXINCDIR="$with_dxdir/include";
+        else
+          DXINCDIR="/usr/local/dx7sdk/include"
+        fi
+
+        if [ $with_dx_libraries ] ; then
+          DXLIBDIR="$with_dx_libraries"
+        elif [ $with_dxdir ] ; then
+          DXLIBDIR="./dx7sdk/lib"
+        else
+          DXLIBDIR="./dx7sdk/lib"
+        fi
+
+        for winapi in $with_winapi
+        do
+        case "$winapi" in
+          directx)
+            echo "DXINCDIR: $DXINCDIR"
+            echo "DXLIBDIR: $DXLIBDIR"
+            OTHER_OBJS="$OTHER_OBJS src/hostapi/dsound/pa_win_ds.o src/hostapi/dsound/pa_win_ds_dynlink.o";
+            LIBS="-ldsound -lole32 $LIBS";
+            DLL_LIBS="${DLL_LIBS} -lm -L$DXLIBDIR -ldsound -lole32";
             #VC98="\"/c/Program Files/Microsoft Visual Studio/VC98/Include\"";
             #CFLAGS="$CFLAGS -I$VC98 -DPA_NO_WMME -DPA_NO_ASIO";
-            CFLAGS="$CFLAGS -I\$(top_srcdir)/include -I$DXDIR/include -DPA_NO_WMME -DPA_NO_ASIO" -DPA_NO_WDMKS;
-        elif [ $with_winapi = "asio" ] ; then
+            CFLAGS="$CFLAGS -I$DXINCDIR"
+            use_ds=yes
+	    need_waveformat=yes
+            ;;
+
+          asio)
             if [ $with_asiodir ] ; then
               ASIODIR="$with_asiodir";
             else
@@ -21313,45 +21359,57 @@
             fi
             echo "ASIODIR: $ASIODIR"
 
-            OTHER_OBJS="pa_asio/pa_asio.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o pa_asio/iasiothiscallresolver.o $ASIODIR/common/asio.o $ASIODIR/host/asiodrivers.o $ASIODIR/host/pc/asiolist.o";
-            LIBS="-lwinmm -lm -lstdc++ -lole32 -luuid";
-            PADLL="portaudio.dll";
-	    THREAD_CFLAGS="-mthreads"
-            SHARED_FLAGS="-shared";
-            DLL_LIBS="${DLL_LIBS} -lwinmm -lm -lstdc++ -lole32 -luuid";
-            CFLAGS="$CFLAGS -ffast-math -fomit-frame-pointer -I\$(top_srcdir)/src/common -I\$(top_srcdir)/pa_asio -I$ASIODIR/host/pc -I$ASIODIR/common -I$ASIODIR/host -DPA_NO_WMME -DPA_NO_DS -DPA_NO_WDMKS -DWINDOWS";
-            CXXFLAGS="$CFLAGS";
-        elif [ $with_winapi = "wdmks" ] ; then
-            if [ $with_dxdir ] ; then
-              DXDIR="$with_dxdir";
-            else
-              DXDIR="/usr/local/dx7sdk";
-            fi
-            echo "DXDIR: $DXDIR"
-            OTHER_OBJS="src/hostapi/wdmks/pa_win_wdmks.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o";
-            LIBS="-lwinmm -lm -luuid -lsetupapi -lole32";
-            PADLL="portaudio.dll";
-	    THREAD_CFLAGS="-mthreads"
-            SHARED_FLAGS="-shared";
-            DLL_LIBS="${DLL_LIBS} -lwinmm -lm -L./dx7sdk/lib -luuid -lsetupapi -lole32";
+            OTHER_OBJS="$OTHER_OBJS pa_asio/pa_asio.o pa_asio/iasiothiscallresolver.o $ASIODIR/common/asio.o $ASIODIR/host/asiodrivers.o $ASIODIR/host/pc/asiolist.o";
+            LIBS="-lstdc++ -lole32 -luuid $LIBS";
+            DLL_LIBS="${DLL_LIBS} -lm -lstdc++ -lole32 -luuid";
+            CFLAGS="$CFLAGS -ffast-math -fomit-frame-pointer -I\$(top_srcdir)/pa_asio -I$ASIODIR/host/pc -I$ASIODIR/common -I$ASIODIR/host -DWINDOWS";
+            CXXFLAGS="$CFLAGS"
+            use_asio=yes
+            ;;
+
+          wdmks)
+            echo "DXINCDIR: $DXINCDIR"
+            echo "DXLIBDIR: $DXLIBDIR"
+            OTHER_OBJS="$OTHER_OBJS src/hostapi/wdmks/pa_win_wdmks.o";
+            LIBS="-luuid -lsetupapi -lole32 $LIBS";
+            DLL_LIBS="${DLL_LIBS} -lm -L$DXLIBDIR -luuid -lsetupapi -lole32";
             #VC98="\"/c/Program Files/Microsoft Visual Studio/VC98/Include\"";
             #CFLAGS="$CFLAGS -I$VC98 -DPA_NO_WMME -DPA_NO_ASIO";
-            CFLAGS="$CFLAGS -I\$(top_srcdir)/src/common -I$DXDIR/include -DPA_NO_WMME -DPA_NO_DS -DPA_NO_ASIO";
-        else   # WMME default
-            OTHER_OBJS="src/hostapi/wmme/pa_win_wmme.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o";
-            LIBS="-lwinmm -lm -lstdc++ -lole32 -luuid";
-            PADLL="portaudio.dll";
-	    THREAD_CFLAGS="-mthreads"
-            SHARED_FLAGS="-shared";
-            DLL_LIBS="${DLL_LIBS} -lwinmm";
-            CFLAGS="$CFLAGS -I\$(top_srcdir)/src/common -DPA_NO_DS -DPA_NO_ASIO -DPA_NO_WDMKS";
+            CFLAGS="$CFLAGS -I$DXINCDIR -DPA_NO_WMME -DPA_NO_DS -DPA_NO_ASIO"
+            use_wdmks=yes
+            ;;
+
+          wmme)
+            OTHER_OBJS="$OTHER_OBJS src/hostapi/wmme/pa_win_wmme.o";
+            LIBS="-lstdc++ -lole32 -luuid $LIBS";
+            use_wmme=yes
+	    need_waveformat=yes
+            ;;
+        esac
+        done
+
+        if [ "$use_ds" != "yes" ]; then
+            CFLAGS="$CFLAGS -DPA_NO_DS"
+        fi
+        if [ "$use_asio" != "yes" ]; then
+            CFLAGS="$CFLAGS -DPA_NO_ASIO"
+        fi
+        if [ "$use_wdmks" != "yes" ]; then
+            CFLAGS="$CFLAGS -DPA_NO_WDMKS"
+        fi
+        if [ "$use_wmme" != "yes" ]; then
+            CFLAGS="$CFLAGS -DPA_NO_WMME"
+        fi
+
+        if [ "$need_waveformat" = "yes" ]; then
+            OTHER_OBJS="$OTHER_OBJS src/os/win/pa_win_waveformat.o"
         fi
         ;;
 
   cygwin* )
 
 	OTHER_OBJS="src/hostapi/wmme/pa_win_wmme.o src/os/win/pa_win_hostapis.o src/os/win/pa_win_util.o";
-	CFLAGS="$CFLAGS -DPA_NO_DS -DPA_NO_WDMKS -DPA_NO_ASIO -DPA_NO_WASAPI"
+	CFLAGS="$CFLAGS -I\$(top_srcdir)/src/os/win -DPA_NO_DS -DPA_NO_WDMKS -DPA_NO_ASIO -DPA_NO_WASAPI"
 	LIBS="-lwinmm -lm";
 	PADLL="portaudio.dll";
 	THREAD_CFLAGS="-mthreads"
@@ -21673,6 +21731,7 @@
 fi
 
 
+        CFLAGS="$CFLAGS -I\$(top_srcdir)/src/os/unix"
 	if [ $have_alsa = "yes" ] && [ $with_alsa != "no" ] ; then
 		DLL_LIBS="$DLL_LIBS -lasound"
 		OTHER_OBJS="$OTHER_OBJS src/hostapi/alsa/pa_linux_alsa.o"

