--- CMakeLists.txt	2013-08-12 00:18:18.000000000 +0300
+++ CMakeLists.txt	2013-08-12 00:53:19.266159894 +0300
@@ -54,7 +54,7 @@
 
 find_package(Gettext)
 find_package(Msgfmt)
-if(Gettext_FOUND AND Msgfmt_FOUND)
+if(GETTEXT_FOUND AND Msgfmt_FOUND)
 	message(STATUS "Localization enabled: Building and installing .mo files")
 	add_subdirectory(lang)
 else()
--- game/CMakeLists.txt	2013-08-11 23:03:02.000000000 +0300
+++ game/CMakeLists.txt	2013-08-11 23:50:15.795391841 +0300
@@ -68,7 +68,8 @@
 list(APPEND LIBS ${Boost_LIBRARIES})
 
 # Find all the libs that don't require extra parameters
-foreach(lib ${OUR_LIBS} SDL2 PangoCairo LibRSVG LibXML++ GLEW AVFormat AVResample SWScale OpenGL Z Jpeg Png PortAudio Fontconfig)
+foreach(lib ${OUR_LIBS} SDL2 PangoCairo LibRSVG LibXML++ GLEW AVFormat AVResample SWScale OpenGL ZLIB JPEG PNG PortAudio Fontconfig)
+
 	find_package(${lib} REQUIRED)
 	include_directories(${${lib}_INCLUDE_DIRS})
 	list(APPEND LIBS ${${lib}_LIBRARIES})
@@ -77,7 +77,7 @@
 endforeach(lib)
 
 find_package(Gettext)
-if(Gettext_FOUND)
+if(GETTEXT_FOUND)
 	include_directories(${Gettext_INCLUDE_DIRS})
 	list(APPEND LIBS ${Gettext_LIBRARIES})
 	add_definitions("-DUSE_GETTEXT")
--- tools/CMakeLists.txt	2014-09-27 11:53:53.000000000 +0300
+++ tools/CMakeLists.txt	2014-09-27 12:19:46.493138788 +0300
@@ -8,7 +8,7 @@
 # Find all the libs that don't require extra parameters
 
 if (APPLE)
-    foreach(lib LibXML++ Z Jpeg Tiff Png Freetype Z)
+    foreach(lib LibXML++ ZLIB JPEG TIFF PNG Freetype)
 	find_package(${lib})
     	if (${lib}_FOUND)
 	    	include_directories(${${lib}_INCLUDE_DIRS})
@@ -21,7 +21,7 @@
     find_package(ImageMagick COMPONENTS Magick++)
     include_directories(${ImageMagick_INCLUDE_DIRS})
 else (APPLE)
-    foreach(lib LibXML++ Z Magick++ Jpeg Tiff Png Freetype Z)
+    foreach(lib LibXML++ ZLIB Magick++ JPEG TIFF PNG Freetype)
 	    find_package(${lib})
     	if (${lib}_FOUND)
 	    	include_directories(${${lib}_INCLUDE_DIRS})
@@ -40,13 +40,13 @@
 if (LibXML++_FOUND)
 	if (Boost_FOUND)
 		if (ImageMagick_FOUND OR Magick++_FOUND) 
-			if (Z_FOUND)
+			if (ZLIB_FOUND)
 				add_executable(ss_extract ss_extract.cc pak.cc ipu_conv.cc ss_cover.cc)
-				target_link_libraries(ss_extract ${LibXML++_LIBRARIES} ${Boost_LIBRARIES} ${Magick++_LIBRARIES} ${Z_LIBRARIES} ${Jpeg_LIBRARIES} ${Tiff_LIBRARIES} ${Png_LIBRARIES} ${Freetype_LIBRARY} ${ImageMagick_LIBRARIES})
+				target_link_libraries(ss_extract ${LibXML++_LIBRARIES} ${Boost_LIBRARIES} ${Magick++_LIBRARIES} ${ZLIB_LIBRARIES} ${Jpeg_LIBRARIES} ${Tiff_LIBRARIES} ${Png_LIBRARIES} ${Freetype_LIBRARY} ${ImageMagick_LIBRARIES})
 				set(targets ${targets} ss_extract)
-			else (Z_FOUND)
+			else (ZLIB_FOUND)
 				message("No zlib found, not building ss_extract")
-			endif (Z_FOUND)
+			endif (ZLIB_FOUND)
 		else (ImageMagick_FOUND OR Magick++_FOUND)
 			message("No Magick++ found, not building ss_extract")
 		endif (ImageMagick_FOUND OR Magick++_FOUND)
@@ -56,7 +56,7 @@
 
 	if (ImageMagick_FOUND OR Magick++_FOUND)
 		add_executable(ss_cover_conv cover_conv.cc pak.cc ss_cover.cc)
-		target_link_libraries(ss_cover_conv ${ImageMagick_LIBRARIES} ${LibXML++_LIBRARIES} ${Z_LIBRARIES} ${Jpeg_LIBRARIES} ${Tiff_LIBRARIES} ${Png_LIBRARIES} ${Freetype_LIBRARY} ${ImageMagick_LIBRARIES} ${Magick++_LIBRARIES})
+		target_link_libraries(ss_cover_conv ${ImageMagick_LIBRARIES} ${LibXML++_LIBRARIES} ${ZLIB_LIBRARIES} ${Jpeg_LIBRARIES} ${Tiff_LIBRARIES} ${Png_LIBRARIES} ${Freetype_LIBRARY} ${ImageMagick_LIBRARIES} ${Magick++_LIBRARIES})
 		set(targets ${targets} ss_cover_conv)
 	else (ImageMagick_FOUND OR Magick++_FOUND)
 		message("No Magick++ found, not building ss_cover_conv")
@@ -65,28 +65,28 @@
 	message("No LibXML++ found, not building ss_extract nor ss_cover_conv")
 endif (LibXML++_FOUND)
 
-if (Boost_FOUND AND Z_FOUND)
+if (Boost_FOUND AND ZLIB_FOUND)
 	add_executable(ss_archive_extract archive_extract.cc)
-	target_link_libraries(ss_archive_extract ${Boost_LIBRARIES} ${Z_LIBRARIES})
+	target_link_libraries(ss_archive_extract ${Boost_LIBRARIES} ${ZLIB_LIBRARIES})
 	set(targets ${targets} ss_archive_extract)
 
 	add_executable(ss_pak_extract pak_extract.cc pak.cc)
-	target_link_libraries(ss_pak_extract ${Boost_LIBRARIES} ${Z_LIBRARIES})
+	target_link_libraries(ss_pak_extract ${Boost_LIBRARIES} ${ZLIB_LIBRARIES})
 	set(targets ${targets} ss_pak_extract)
 
 	add_executable(itg_pck itg_pck.cc)
-	target_link_libraries(itg_pck ${Boost_LIBRARIES} ${Z_LIBRARIES})
+	target_link_libraries(itg_pck ${Boost_LIBRARIES} ${ZLIB_LIBRARIES})
 	set(targets ${targets} itg_pck)
 
 	add_executable(ss_chc_decode ss_chc_decode.cc)
-	target_link_libraries(ss_chc_decode ${Boost_LIBRARIES} ${Z_LIBRARIES})
+	target_link_libraries(ss_chc_decode ${Boost_LIBRARIES} ${ZLIB_LIBRARIES})
 	set(targets ${targets} ss_chc_decode)
 
 endif()
 
-if (Z_FOUND)
+if (ZLIB_FOUND)
 	add_executable(ss_adpcm_decode adpcm_decode.cc pak.cc)
-	target_link_libraries(ss_adpcm_decode ${Z_LIBRARIES})
+	target_link_libraries(ss_adpcm_decode ${ZLIB_LIBRARIES})
 	set(targets ${targets} ss_adpcm_decode)
 endif()
 
