diff -urN Performous-0.6.1-Source.old/cmake/Modules/FindAVUtil.cmake Performous-0.6.1-Source/cmake/Modules/FindAVUtil.cmake
--- Performous-0.6.1-Source.old/cmake/Modules/FindAVUtil.cmake	2012-05-16 13:44:26.112930173 +0200
+++ Performous-0.6.1-Source/cmake/Modules/FindAVUtil.cmake	2012-05-16 13:56:10.229914635 +0200
@@ -25,9 +25,15 @@
       if(EXISTS "${AVUtil_INCLUDE_DIR}/${suffix}avutil.h")
         set(AVUtil_INCLUDE "${suffix}avutil.h")
       endif(EXISTS "${AVUtil_INCLUDE_DIR}/${suffix}avutil.h")
+      if(EXISTS "${AVUtil_INCLUDE_DIR}/${suffix}mathematics.h")
+        set(AVUtilMath_INCLUDE "${suffix}mathematics.h")
+      endif(EXISTS "${AVUtil_INCLUDE_DIR}/${suffix}mathematics.h")
     endif(NOT AVUtil_INCLUDE)
   endforeach(suffix)
 
+  if(NOT AVUtilMath_INCLUDE)
+    message(FATAL_ERROR "Found mathematics.h include dir, but not the header file. Perhaps you need to clear CMake cache?")
+  endif(NOT AVUtilMath_INCLUDE) 
   if(NOT AVUtil_INCLUDE)
     message(FATAL_ERROR "Found avutil.h include dir, but not the header file. Perhaps you need to clear CMake cache?")
   endif(NOT AVUtil_INCLUDE)
diff -urN Performous-0.6.1-Source.old/game/config.cmake.hh Performous-0.6.1-Source/game/config.cmake.hh
--- Performous-0.6.1-Source.old/game/config.cmake.hh	2012-05-16 13:44:26.103930173 +0200
+++ Performous-0.6.1-Source/game/config.cmake.hh	2012-05-16 13:59:24.552910345 +0200
@@ -15,6 +15,7 @@
 #define AVCODEC_INCLUDE <@AVCodec_INCLUDE@>
 #define AVFORMAT_INCLUDE <@AVFormat_INCLUDE@>
 #define SWSCALE_INCLUDE <@SWScale_INCLUDE@>
+#define AVUTILMATH_INCLUDE <@AVUtilMath_INCLUDE@>
 
 #endif
 
diff -urN Performous-0.6.1-Source.old/game/ffmpeg.cc Performous-0.6.1-Source/game/ffmpeg.cc
--- Performous-0.6.1-Source.old/game/ffmpeg.cc	2012-05-16 13:44:26.101930173 +0200
+++ Performous-0.6.1-Source/game/ffmpeg.cc	2012-05-16 14:00:38.278908717 +0200
@@ -10,6 +10,7 @@
 #include AVCODEC_INCLUDE
 #include AVFORMAT_INCLUDE
 #include SWSCALE_INCLUDE
+#include AVUTILMATH_INCLUDE
 }
 
 // #define USE_FFMPEG_CRASH_RECOVERY
