--- game/CMakeLists.txt	2013-08-10 23:34:20.000000000 +0300
+++ game/CMakeLists.txt	2013-08-11 00:10:47.073738719 +0300
@@ -150,7 +150,7 @@
 # Build main executable
 add_executable(performous ${SUBSYSTEM_WIN32} ${SOURCES} ${SDL_SOURCES} ${RESOURCE_OBJECT})
 target_link_libraries(performous ${LIBS})
-install(TARGETS performous DESTINATION ${BIN_INSTALL})
+install(TARGETS performous DESTINATION @GENTOO_BINDIR@)
 
 set_target_properties(performous PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE)  # Store library paths in executable
 set_target_properties(performous PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})  # Produce executable in build/, not build/game/
