--- lang/CMakeLists.txt	2014-09-27 12:29:44.000000000 +0300
+++ lang/CMakeLists.txt	2014-09-27 13:51:39.908581638 +0300
@@ -1,13 +1,12 @@
 cmake_minimum_required(VERSION 2.6)
-file(GLOB LANGUAGES *.po)
-foreach(language ${LANGUAGES})
-	string(REGEX REPLACE "(.+(\\\\|/))+" "" language ${language})
-	string(REGEX REPLACE "\\.po$" "" language ${language})
+foreach(language $ENV{LINGUAS})
 	set(pofile "${CMAKE_CURRENT_SOURCE_DIR}/${language}.po")
-	set(mobuild "${CMAKE_CURRENT_BINARY_DIR}/${language}/LC_MESSAGES")
-	set(mofile "${mobuild}/${CMAKE_PROJECT_NAME}.mo")
-	file(MAKE_DIRECTORY "${mobuild}")
-	add_custom_command(OUTPUT "${mofile}" COMMAND "${Msgfmt_BIN}" -v "${pofile}" -o "${mofile}" MAIN_DEPENDENCY "${pofile}" COMMENT "Building ${language} locale" VERBATIM)
-	add_custom_target("locale_${language}" ALL DEPENDS "${mofile}")  # Make sure the mofiles are always built
-	install(FILES "${mofile}" DESTINATION "${LOCALE_DIR}/${language}/LC_MESSAGES")
+	if (EXISTS ${pofile})
+		set(mobuild "${CMAKE_CURRENT_BINARY_DIR}/${language}/LC_MESSAGES")
+		set(mofile "${mobuild}/${CMAKE_PROJECT_NAME}.mo")
+		file(MAKE_DIRECTORY "${mobuild}")
+		add_custom_command(OUTPUT "${mofile}" COMMAND "${Msgfmt_BIN}" -v "${pofile}" -o "${mofile}" MAIN_DEPENDENCY "${pofile}" COMMENT "Building ${language} locale" VERBATIM)
+		add_custom_target("locale_${language}" ALL DEPENDS "${mofile}")  # Make sure the mofiles are always built
+		install(FILES "${mofile}" DESTINATION "${LOCALE_DIR}/${language}/LC_MESSAGES")
+	endif()
 endforeach(language)
